Add documentation for testing with local Playground packages

bcotrim opened the pull request:
## Related issues

- N/A

## Proposed Changes

- Add new documentation file `docs/testing-with-local-playground.md` with step-by-step guide for testing Studio with locally built WordPress Playground packages
- Update `CONTRIBUTING.md` to link to the new documentation

## Testing Instructions

- Follow the documentation steps to set up local Playground packages
- Verify Studio runs with the local packages
- Make a change to a Playground package and confirm it's reflected in Studio

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

bcotrim committed:
Add documentation for testing with local Playground packages

wojtekn commented on docs/testing-with-local-playground.md:
````diff
@@ -0,0 +1,110 @@
+```json
+{
+  "dependencies": {
+    "@php-wasm/fs-journal": "file:../wordpress-playground/dist/packages/php-wasm/fs-journal",
````
Would it also make sense to reorganize nodes in the `package.json` to make the replacement easier?

wojtekn commented on docs/testing-with-local-playground.md:
````diff
@@ -0,0 +1,110 @@
+Run this from the wordpress-playground root:
+
+```bash
+cd /path/to/wordpress-playground
````
What if we added this script to `tools/scripts/` in the Playground repository, and made it available as NPM command in `package.json`? 

@adamziel any thoughts?

wojtekn replied:
@akirk what do you think?

wojtekn commented on docs/testing-with-local-playground.md:
````diff
@@ -0,0 +1,110 @@
+   npx nx build playground-cli  # or other package name
+   ```
+
+2. Restart Studio (type `rs` in the terminal or restart `npm start`)
````
I couldn't make the `rs` command work, and I needed to restart `npm start`.

wojtekn commented on docs/testing-with-local-playground.md:
```diff
@@ -0,0 +1,110 @@
+
+To go back to using the published npm packages:
+
+1. Restore the original version numbers in `package.json` (use `git checkout package.json`)
```
I needed to revert `package-lock.json`, too. In the other case, it kept installing symlinked packages.

wojtekn commented on docs/testing-with-local-playground.md:
```diff
@@ -0,0 +1,110 @@
+
+## Reverting to npm Packages
+
+To go back to using the published npm packages:
```
Should we also include instructions showing how to revert changes in Playground repo?

wojtekn reviewed (commented):
This is a great start! I was able to run dev-mode Playground in Studio, easily modify it, and apply changes in Studio:

```
[playground-cli] Enabling experimental multi-worker support with 13 workers (CPU cores - 1)
[playground-cli] The fs-ext package is not installed. Internal file locking will not be integrated with host OS file locking.
[playground-cli] Starting a PHP server...
[playground-cli] Starting up workers
[playground-cli] Skipping YOLO SQLite integration plugin setup...
[playground-cli] Booting WordPress...
```

bcotrim committed:
Address PR review comments for local Playground testing docs

bcotrim commented:
Thanks for the review @wojtekn! I've addressed the comments:
  - Grouped Playground packages at the end of package.json for easier replacement
  - Removed the rs command reference
  - Updated revert instructions to include package-lock.json and Playground repo

  Can you take another look?

wojtekn commented on package.json:
```diff
@@ -158,7 +152,14 @@
 		"yauzl": "^3.2.0",
-		"zod": "^3.24.3"
+		"zod": "^3.24.3",
+
```
Looks good, but we can remove the empty line.

bcotrim committed:
Remove empty line before Playground packages
